packages/core/src/transforms/resize.ts

```diff
@@ -73,13 +83,26 @@ export const resize: TransformFactory<ResizeOptions> = (config) => {
       finalWidth = height * (aspect || originalAspect)
     }
 
+    if (
+      (withoutEnlargement && (finalHeight > originalHeight || finalWidth > originalWidth)) ||
+      (withoutReduction && (finalHeight < originalHeight || finalWidth < originalWidth))
+    ) {
+      finalHeight = originalHeight
```

Comment:
I might be worth adding a `console.warn` here to let the user know that the `withoutEnlargement` or `withoutReduction` case got hit, otherwise it might be surprising that the resize "didn't work"

Reply:
I couldn't find any other warnings in the app beyond the tests, so I've added one prefixed with `[vite-imagetools]`. Not sure if you have a standard you're looking for here